java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -767,4 +792,62 @@ public String getClientLibToken() {
   public String getQuotaProjectId() {
     return quotaProjectId;
   }
+
+  /**
+   * Returns the resolved endpoint for the Service to connect to Google Cloud
+   *
+   * <p>The resolved endpoint is always in `host:port` format
+   */
+  public String getResolvedEndpoint(String serviceName) {
+    if (universeDomain == null) {
+      return formatEndpoint(serviceName, Credentials.GOOGLE_DEFAULT_UNIVERSE);
+    } else if (universeDomain.isEmpty()) {
+      throw new IllegalArgumentException("Universe Domain cannot be empty");
+    } else {
+      if (host != null) {
```

Comment:
If host is set and universeDomain is null, I think we should use `normalizeEndpoint()` instead of `formatEndpoint(serviceName, Credentials.GOOGLE_DEFAULT_UNIVERSE)`?